portal-api/api-tests/v1/gef/application-submit.api-test.js

```diff
@@ -197,3 +200,45 @@ describe('removeChangedToIssued()', () => {
     expect(changedToIssuedValue).toEqual(false);
   });
 });
+
+describe('checkCoverDateConfirmed()', () => {
+  let mockApplication;
+  let aMaker;
+
+  // Create facilities with issued and unissued stages
+  beforeAll(async () => {
+    const testUsers = await testUserCache.initialise(app);
+    aMaker = testUsers().withRole('maker').one();
+    const mockAIN = mockApplications[0];
+    mockApplication = await as(aMaker).post(mockAIN).to(applicationBaseUrl);
+    mockApplication = await as(aMaker).put({ submissionType: CONSTANTS.DEAL.SUBMISSION_TYPE.AIN }).to(`${applicationBaseUrl}/${mockApplication.body._id}`);
+
+    await wipeDB.wipe([collectionName]);
+    await wipeDB.wipe([applicationCollectionName]);
+
+    await as(aMaker).post({
+      dealId: mockApplication.body._id,
+      type: FACILITY_TYPE.CASH,
+      hasBeenIssued: true,
+      coverDateConfirmed: null,
+    }).to(baseUrl);
+
+    await as(aMaker).post({
+      dealId: mockApplication.body._id,
+      type: FACILITY_TYPE.CASH,
+      hasBeenIssued: false,
+      coverDateConfirmed: null,
+    }).to(baseUrl);
+
+    await as(aMaker).post({
+      dealId: mockApplication.body._id,
+      type: FACILITY_TYPE.CASH,
+      hasBeenIssued: true,
+      coverDateConfirmed: false,
+    }).to(baseUrl);
+  });
+
+  it('Should return `2` when the application status is DRAFT, type is AIN and have atleast one facility with issued status', async () => {
+    expect(await checkCoverDateConfirmed(mockApplication.body)).toEqual(2);
+  });
+});
```

Comment:
Can we add a test for it returning 0 when nothing is updated?

Reply:
Done, have added more test cases covering all possible scenarios.